Add key to client

bun919tw opened the pull request:
Currently client will only have one instance in the whole process. However, client should be changed to be stateful due to future db implementation. Therefore, add one variable named `key` to client constructor indicating the owner of the client.

boolafish commented on plasma_cash/client/client.py:
```diff
@@ -9,27 +9,28 @@
 
-    def deposit(self, amount, depositor, currency):
+    def deposit(self, amount, currency):
+        depositor = utils.privtoaddr(self.key)
```
`utils.privtoaddr` is used everywhere. 
Probably we can add another field or use `@property` inside client for `address` and use `self.address` instead?

bun919tw replied:
Sure!

boolafish commented on unit_tests/client/test_client.py:
```diff
@@ -21,25 +21,33 @@ def child_chain(self):
     @pytest.fixture(scope='function')
     def client(self, root_chain, child_chain):
-        return Client(root_chain, child_chain)
+        DUMMY_KEY = 'key'
```
Minor: I know I am the one that start the style of mocking everything. However, if by using some real key value and we can avoid those mocks on eg. `when('plasma_cash.client.client.utils').privtoaddr(client.key).thenReturn(DUMMY_XXX)`, it might be easier to read the test and the test also covers better. And the effort of using real key is not that hard as well (supposedly). However, this is just my opinion (recently got). Would leave this to you.

bun919tw commented:
@boolafish I have
1. Added one property named `address` in the client.
2. Changed the `key` to a real one in client unit test.

boolafish commented on unit_tests/client/test_client.py:
```diff
@@ -21,41 +20,48 @@ def child_chain(self):
+        client.deposit(DUMMY_AMOUNT, DUMMY_CURRENCY)
 
-        verify(MOCK_TRANSACT).transact(ANY)
+        verify(MOCK_TRANSACT).transact({'from': DUMMY_ADDRESS, 'value': DUMMY_AMOUNT * 10**18})
```
I'm actually thinking we can just use `client.address`? and we don't need to declare DUMMY_ADDRESS anymore?

bun919tw commented:
@boolafish Thanks for the review. I've removed the dummy address in client unit tests and I also changed the title of this PR and commit.
Merge this PR by myself.
